tests/test_cache.py

```diff
@@ -1,504 +1,372 @@
-        params = {"a": "a", "b": ["1", "2", "3"], "c": "4"}
-
-        self.assertFalse(self.s.get(url, params=params).from_cache)
-        r = self.s.get(url, params=params)
-        self.assertTrue(r.from_cache)
-        self.assertEqual(r.json()["args"], params)
-        self.assertFalse(self.s.get(url, params={"a": "b"}).from_cache)
-        self.assertTrue(self.s.get(url, params=sorted(params.items())).from_cache)
-
-        class UserSubclass(dict):
-            def items(self):
-                return sorted(super(UserSubclass, self).items(), reverse=True)
-
-        params["z"] = "5"
-        custom_dict = UserSubclass(params)
-        self.assertFalse(self.s.get(url, params=custom_dict).from_cache)
-        self.assertTrue(self.s.get(url, params=custom_dict).from_cache)
-
-    def test_post_parameters_normalization(self):
-        params = {"a": "a", "b": ["1", "2", "3"], "c": "4"}
-        url = httpbin("post")
-        s = CachedSession(CACHE_NAME, CACHE_BACKEND, allowable_methods=('GET', 'POST'))
-        self.assertFalse(s.post(url, data=params).from_cache)
-        self.assertTrue(s.post(url, data=params).from_cache)
-        self.assertTrue(s.post(url, data=sorted(params.items())).from_cache)
-        self.assertFalse(s.post(url, data=sorted(params.items(), reverse=True)).from_cache)
-
-    def test_stream_requests_support(self):
-        n = 100
-        url = httpbin("stream/%s" % n)
-        r = self.s.get(url, stream=True)
-        first_char = r.raw.read(1)
-        lines = list(r.iter_lines())
-        self.assertTrue(first_char)
-        self.assertEqual(len(lines), n)
+        for i in range(5):
+            r = mock_session.get(httpbin('get'), hooks={hook: hook_func})
+        assert state[hook] == 5
+
+
+def test_normalize_params(mock_session):
+    params = {"a": "a", "b": ["1", "2", "3"], "c": "4"}
+
+    assert mock_session.get(MOCKED_URL, params=params).from_cache is False
+    assert mock_session.get(MOCKED_URL, params=params).from_cache is True
+    assert mock_session.get(MOCKED_URL, params={"a": "b"}).from_cache is False
+    assert mock_session.get(MOCKED_URL, params=dict(sorted(params.items()))).from_cache is True
+
+    class UserSubclass(dict):
+        def items(self):
+            return sorted(super(UserSubclass, self).items(), reverse=True)
+
+    params["z"] = "5"
+    custom_dict = UserSubclass(params)
+    assert mock_session.get(MOCKED_URL, params=custom_dict).from_cache is False
+    assert mock_session.get(MOCKED_URL, params=custom_dict).from_cache is True
+
+
+def test_normalize_post_data(mock_session):
+    params = {"a": "a", "b": ["1", "2", "3"], "c": "4"}
+    assert mock_session.post(MOCKED_URL, data=params).from_cache is False
+    assert mock_session.post(MOCKED_URL, data=params).from_cache is True
+    assert mock_session.post(MOCKED_URL, data=sorted(params.items())).from_cache is True
+    assert mock_session.post(MOCKED_URL, data=sorted(params.items(), reverse=True)).from_cache is False
```

Comment:
This surprised me, but I guess the dictionary is converted to a string before checking the cache. Maybe having some text data instead of a dictionary might make sense here as well (and the behavior more obvious).

Reply:
Yeah, I left this test and the one above it are pretty much the same as before this PR. I didn't add any more tests for this yet because there are several other things that related to request normalization that need to be handled in a separate PR. See issue #80